apps/andi/lib/andi/harvest/data_json_dataset_mapper.ex

```diff
@@ -0,0 +1,83 @@
+defmodule Andi.Harvest.DataJsonToDataset do
+  @moduledoc """
+  maps data json to %SmartCity.Dataset{}
+  """
+  alias Andi.InputSchemas.Datasets
+
+  @scos_data_json_uuid "1719bf64-38f5-40bf-9737-45e84f5c8419"
+
+  def mapper(%{"dataset" => datasets}, org) do
+    Enum.map(datasets, fn data_json_dataset -> dataset_model(data_json_dataset, org) end)
+    |> Enum.filter(&is_public?/1)
+    |> Enum.reject(&Enum.empty?/1)
+    |> Enum.reduce([], fn dataset, acc ->
+      case SmartCity.Dataset.new(dataset) do
+        {:ok, dataset} -> [dataset | acc]
+        _ -> acc
+      end
+    end)
+  end
+
+  defp dataset_model(data_json_dataset, org) do
+    %{
+      "business" => %{
+        "categories" => data_json_dataset["theme"],
+        "conformsToUri" => "https://project-open-data.cio.gov/v1.1/schema/",
+        "contactEmail" => Map.get(data_json_dataset["contactPoint"], :hasEmail),
+        "contactName" => Map.get(data_json_dataset["contactPoint"], :fn),
+        "dataTitle" => data_json_dataset["title"],
+        "describedByMimeType" => data_json_dataset["describedByType"],
+        "describedByUrl" => data_json_dataset["describedBy"],
+        "description" => data_json_dataset["description"],
+        "homepage" => data_json_dataset["landingPage"],
+        "issuedDate" => data_json_dataset["issued"],
+        "keywords" => data_json_dataset["keyword"],
+        "language" => data_json_dataset["language"],
+        "license" => data_json_dataset["license"],
+        "modifiedDate" => data_json_dataset["modified"],
+        "orgTitle" => org.orgTitle,
+        "parentDataset" => data_json_dataset["isPartOf"],
+        "referenceUrls" => data_json_dataset["references"],
+        "rights" => data_json_dataset["rights"],
+        "spatial" => data_json_dataset["spatial"]
+      },
+      "id" => UUID.uuid5(@scos_data_json_uuid, data_json_dataset["identifier"]),
+      "technical" => %{
+        "dataName" => data_name(data_json_dataset["title"]),
+        "orgName" => org.orgName,
+        "orgId" => org.id,
+        "sourceFormat" => "text/html",
+        "private" => access_level(data_json_dataset["accessLevel"]),
+        "sourceType" => "remote",
+        "sourceUrl" => source_url(data_json_dataset["distribution"]),
+        "systemName" => system_name(org.orgName, data_name(data_json_dataset["title"]))
+      }
+    }
+  end
+
+  defp access_level(access_level) do
+    case access_level do
+      "non-public" -> true
+      "public" -> false
```

Comment:
Do we need to check "public" -> false here. I think it can be handled in default case